lib/python/picongpu/plugins/plot_mpl/emittance_evolution_visualizer.py

```diff
@@ -0,0 +1,180 @@
+            sim_name is a short string used e.g. in plot legends.
+            sim_path leads to the run directory of PIConGPU
+            (the path before ``simOutput/``).
+            If None, the user is responsible for providing run_directories
+            later on via set_run_directories() before calling visualize().
+        ax: matplotlib.axes
+        """
+        super().__init__(EmittanceData, run_directories, ax)
+        self.plt_lin = None  # plots line at current itteration
+        self.cur_iteration = None
+
+    def _create_plt_obj(self, idx):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        emit, y_slices, all_iterations, dt = self.data[idx]
+        label = self.sim_labels[idx]
+        np_data = np.zeros(len(all_iterations))
+        for index, ts in enumerate(all_iterations):
+            np_data[index] = emit[index][0]
+        ps = 1.e12  # for conversion from s to ps
+        # np_data * 1.e6 converts emittance to pi mm mrad
+        self.plt_obj[idx], = self.ax.plot(all_iterations * dt * ps,
+                                          np_data * 1.e6, scalex=True,
+                                          scaley=True, label=label,
+                                          color=self.colors[idx])
+        if self.cur_iteration:
+            self.plt_lin = self.ax.axvline(self.cur_iteration * dt * ps,
+                                           color='#FF6600')
+
+    def _update_plt_obj(self, idx):
+        """
+        Implementation of base class function.
+        """
+        emit, y_slices, all_iterations, dt = self.data[idx]
+        np_data = np.zeros(len(all_iterations))
+        for index, ts in enumerate(all_iterations):
+            np_data[index] = emit[index][0]
+        if self.plt_lin:
+            self.plt_lin.remove()
+        ps = 1.e12  # for conversion from s to ps
+        # np_data * 1.e6 converts emittance to pi mm mrad
+        self.plt_obj[idx].set_data(all_iterations * dt * ps, np_data * 1.e6)
+        if self.cur_iteration:
+            self.plt_lin = self.ax.axvline(self.cur_iteration * dt * ps,
+                                           color='#FF6600')
+
+    def visualize(self, **kwargs):
+        """
+        Creates a semilogy plot on the provided axes object for
+        the data of the given iteration using matpotlib.
+
+        Parameters
+        ----------
+        kwargs: dictionary with further keyword arguments, valid are:
+            species: string
+                short name of the particle species, e.g. 'e' for electrons
+                (defined in ``speciesDefinition.param``)
+            iteration: int
+                number of the iteration
+            species_filter: string
+                name of the particle species filter, default is 'all'
+                (defined in ``particleFilters.param``)
+
+        """
+        self.cur_iteration = kwargs.get('iteration')
```

Comment:
I just realized since `kwargs` is a `dict` and `dict.get()` returns `None` in Python this is already optional. Good! Just he `-i iteration` below needs then fixing.